### ISSUE TITLE
Error configuring IDS due to syntax in policy Description "error installing ids rules (Error (1))"

### ISSUE THREAD

**Issue by rchenzheng** [opnsense/core#4946]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [x] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [x] I have searched the existing issues and I am convinced that mine is new.

**Describe the bug**

IPS/IDS policies description are not being escaped and provides no syntax/format checking, when a percentage sign is put in the error gives no indication of a syntax problem. In the edit screen, the description should prevent the user from inputting symbols or should at least be escaped.

For example: 

`Drop and 50% confidence`

```
"error installing ids rules (Error (1))"
```

**To Reproduce**

1. Go to 'Services: Intrusion Detection: Policy'
2. Click on '+'
3. Create a rule with Description: `Drop and 50% confidence` 
4. Click Apply

**Expected behavior**

In the edit screen or on apply, it should escape the symbols or special characters instead of throwing this error.

Or it should tell the user to fix the syntax of the `Description` in the edit screen

**Describe alternatives you considered**

Not to use special characters such as `percentage %`

**Screenshots**

**Relevant log files**

Also other red herrings which clobbered by troubleshooting steps

```
Apr 26 11:05:28 OPNsense suricata[53059]: [100779] <Notice> -- This is Suricata version 5.0.6 RELEASE running in SYSTEM mode
Apr 26 11:06:07 OPNsense suricata[89662]: [100228] <Warning> -- [ERRCODE: SC_WARN_FLOWBIT(306)] - flowbit 'is_proto_irc' is checked but not set. Checked in 2002029 and 4 other sigs
Apr 26 11:06:07 OPNsense suricata[89662]: [100228] <Warning> -- [ERRCODE: SC_WARN_FLOWBIT(306)] - flowbit 'ET.http.javaclient.vulnerable' is checked but not set. Checked in 2013036 and 0 other sigs
Apr 26 11:06:07 OPNsense suricata[89662]: [100228] <Warning> -- [ERRCODE: SC_WARN_FLOWBIT(306)] - flowbit 'ET.ELFDownload' is checked but not set. Checked in 2019896 and 0 other sigs
Apr 26 11:06:07 OPNsense suricata[89662]: [100228] <Warning> -- [ERRCODE: SC_WARN_FLOWBIT(306)] - flowbit 'et.DocVBAProject' is checked but not set. Checked in 2020170 and 0 other sigs
Apr 26 11:06:07 OPNsense suricata[89662]: [100228] <Warning> -- [ERRCODE: SC_WARN_FLOWBIT(306)] - flowbit 'ET.MSSQL' is checked but not set. Checked in 2020569 and 0 other sigs
Apr 26 11:06:07 OPNsense suricata[89662]: [100228] <Warning> -- [ERRCODE: SC_WARN_FLOWBIT(306)] - flowbit 'ETPRO.RTF' is checked but not set. Checked in 2020700 and 0 other sigs
Apr 26 11:06:07 OPNsense suricata[89662]: [100228] <Warning> -- [ERRCODE: SC_WARN_FLOWBIT(306)] - flowbit 'et.MCOFF' is checked but not set. Checked in 2022303 and 0 other sigs
Apr 26 11:06:07 OPNsense suricata[89662]: [100228] <Warning> -- [ERRCODE: SC_WARN_FLOWBIT(306)] - flowbit 'ET.autoit.ua' is checked but not set. Checked in 2019165 and 0 other sigs
Apr 26 11:06:07 OPNsense suricata[89662]: [100228] <Warning> -- [ERRCODE: SC_WARN_FLOWBIT(306)] - flowbit 'et.IE7.NoRef.NoCookie' is checked but not set. Checked in 2023671 and 4 other sigs
Apr 26 11:06:07 OPNsense suricata[89662]: [100228] <Warning> -- [ERRCODE: SC_WARN_FLOWBIT(306)] - flowbit 'min.gethttp' is checked but not set. Checked in 2023711 and 0 other sigs
Apr 26 11:06:07 OPNsense suricata[89662]: [100228] <Warning> -- [ERRCODE: SC_WARN_FLOWBIT(306)] - flowbit 'ET.http.binary' is checked but not set. Checked in 2023741 and 4 other sigs
Apr 26 11:06:07 OPNsense suricata[89662]: [100228] <Warning> -- [ERRCODE: SC_WARN_FLOWBIT(306)] - flowbit 'ET.armwget' is checked but not set. Checked in 2024241 and 1 other sigs
Apr 26 11:06:07 OPNsense suricata[89662]: [100228] <Warning> -- [ERRCODE: SC_WARN_FLOWBIT(306)] - flowbit 'ET.smb.binary' is checked but not set. Checked in 2027402 and 4 other sigs
Apr 26 11:06:07 OPNsense suricata[89662]: [100228] <Warning> -- [ERRCODE: SC_WARN_FLOWBIT(306)] - flowbit 'ET.Socks5.OnionReq' is checked but not set. Checked in 2027704 and 0 other sigs
Apr 26 11:06:07 OPNsense suricata[89662]: [100228] <Warning> -- [ERRCODE: SC_WARN_FLOWBIT(306)] - flowbit 'ET.http.javaclient' is checked but not set. Checked in 2017557 and 1 other sigs
Apr 26 11:06:07 OPNsense suricata[89662]: [100228] <Warning> -- [ERRCODE: SC_WARN_FLOWBIT(306)] - flowbit 'et.JavaArchiveOrClass' is checked but not set. Checked in 2017772 and 1 other sigs
Apr 26 11:06:07 OPNsense suricata[89662]: [100228] <Warning> -- [ERRCODE: SC_WARN_FLOWBIT(306)] - flowbit 'ET.pdf.in.http' is checked but not set. Checked in 2017790 and 0 other sigs
Apr 26 11:07:21 OPNsense suricata[89662]: [100335] <Notice> -- opened netmap:igb0/R from igb0: 0x57b6b3b3000
Apr 26 11:07:21 OPNsense suricata[89662]: [100335] <Notice> -- opened netmap:igb0^ from igb0^: 0x57b6b3b3300
Apr 26 11:07:21 OPNsense suricata[89662]: [100753] <Notice> -- opened netmap:igb0^ from igb0^: 0x57b6febf000
Apr 26 11:07:21 OPNsense suricata[89662]: [100753] <Notice> -- opened netmap:igb0/T from igb0: 0x57b6febf300
Apr 26 11:07:21 OPNsense suricata[89662]: [100228] <Notice> -- all 2 packet processing threads, 4 management threads initialized, engine started.
```

**Additional context**

After seeing this error, it makes it seem as if the IPS/IDS isn't functioning correctly as rules are not checked under 

Services: Intrusion Detection: Administration -> rules

**Environment**

Software version used and hardware type if relevant, e.g.:

- OPNsense 21.1.5-amd64
- FreeBSD 12.1-RELEASE-p16-HBSD
- OpenSSL 1.1.1k 25 Mar 2021

Intel Kaby Lake R i5-8250U (64 Bit, 1.6GHz base, 3.4GHz Turbo, 4 Cores, 8 Threads, 6MB Intel Smart Cache)
Network Intel® I210

**Comment by Xboarder56** [comment]
Just observed this as well. This is the log I have from the same error:

```
[d9d68178-e98c-4628-b104-a8f1391bfbeb] Script action failed with Command '/usr/local/opnsense/scripts/suricata/listRuleMetadata.py' returned non-zero exit status 1. at Traceback (most recent call last): File "/usr/local/opnsense/service/modules/processhandler.py", line 478, in execute subprocess.check_call(script_command, env=self.config_environment, shell=True, File "/usr/local/lib/python3.8/subprocess.py", line 364, in check_call raise CalledProcessError(retcode, cmd) subprocess.CalledProcessError: Command '/usr/local/opnsense/scripts/suricata/listRuleMetadata.py' returned non-zero exit status 1.
```

**Comment by AdSchellevis** [comment]
pt research? https://github.com/opnsense/core/pull/5148

**Comment by Xboarder56** [comment]
> pt research? #5148

Yep, that’s probably it

**Comment by ilikenwf** [comment]
PT-Research is causing this issue for me as well.

**Comment by AdSchellevis** [comment]
incomplete metadata, fixed with https://github.com/opnsense/core/commit/f6817b009efaf571ab73b86e112bd1584ee7004a. let's close this issue, people can patch existing installations or wait until a release with the fix in is released (probably 21.7.2)
